app/controllers/companies_controller.rb

```diff
@@ -1,7 +1,17 @@
 class CompaniesController < ApplicationController
+
   def index
-    @companies = Company.all
-  end
+      @company = Company.where(city: params[:location])
+      @companies = Company.all.order(:city)
+      @jobs = Company.where(city: params[:location])
```

Comment:
we only need to grab this data if we have location in the `params` same with the sorted list of companies by city. only set the instance variables that you need